rclcpp/test/rclcpp/test_executor.cpp

```diff
@@ -0,0 +1,450 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <memory>
+#include <stdexcept>
+#include <string>
+
+#include "rclcpp/executor.hpp"
+#include "rclcpp/memory_strategy.hpp"
+#include "rclcpp/strategies/allocator_memory_strategy.hpp"
+
+#include "../mocking_utils/patch.hpp"
+#include "../utils/rclcpp_gtest_macros.hpp"
+
+class DummyExecutor : public rclcpp::Executor
```

Comment:
You didn't want to add these tests to test_executors.cpp?

Reply:
Hmpf.  Well, the whole executor/executors naming thing in this repository somewhat confuses me.  I wanted to make this file match the `.cpp` file it was testing.  Also, adding it to `test_executors.cpp` would make that file really long.  But I actually don't feel strongly either way, so if you think they should be combined, I can do that.

Reply:
The main thing about adding it to test_executors.cpp is that those tests are parameterized against all executors. But these tests you're adding are against a generic executor and testing the base class code. That would be the biggest difference. 

Since you're not using a specific executor here and instead just the base class code, then I think it's fine to leave it as is. I was mainly just curious. I do think you should add a disclaimer comment at the top describing this file is only intended to test the base class and everything else should be considered for specific executor test files or the parameterized test file.

Reply:
Sounds good, I'll go ahead and add the disclaimer.

Reply:
(I did this in the latest, where I also rebased)